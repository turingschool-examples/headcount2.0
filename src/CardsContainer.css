.cards-container {
  display: grid;
  grid-gap: 5px;
  grid-column: 2;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
}

