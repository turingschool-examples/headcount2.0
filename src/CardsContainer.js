import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const CardsContainer = ({ districts }) => {
  return (
    <div> 
      {
        Object.keys(districts.stats)
          .map( (district, index) => 
            <Card 
              district={districts.stats[district]}
              key={index}
            />
          )
      }
    </div>
  );
};

CardsContainer.propTypes = { 
  districts: PropTypes.object.isRequired
};

export default CardsContainer;