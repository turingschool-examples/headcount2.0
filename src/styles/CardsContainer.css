.CardsContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  margin: auto;
  width: 85%;
}