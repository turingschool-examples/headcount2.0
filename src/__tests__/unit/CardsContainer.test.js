import React from 'react';
import CardsContainer from '../../CardsContainer';
import {shallow} from 'enzyme';

describe('CardsContainer', () => {
	let districts = {stats: {
		COLORADO: {
			location: 'COLORADO',
			stats: {
				'2017': 0
			}
		}, 
		DENVER: {
			location: 'DENVER',
			stats: {
				'2017': 0
			}
		}
	}
}

	it('should have array of objects as props', () => {
		let cardsContainer = shallow(<CardsContainer districts={districts} />);
		expect(cardsContainer).toMatchSnapshot();
		expect(cardsContainer.props("districts")).toBeDefined();
	});

})