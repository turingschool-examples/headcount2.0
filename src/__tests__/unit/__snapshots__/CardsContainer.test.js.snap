// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsContainer should have array of objects as props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardsContainer
    districts={
        Object {
            "stats": Object {
              "COLORADO": Object {
                "location": "COLORADO",
                "stats": Object {
                  "2017": 0,
                },
              },
              "DENVER": Object {
                "location": "DENVER",
                "stats": Object {
                  "2017": 0,
                },
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          district={
                    Object {
                              "location": "COLORADO",
                              "stats": Object {
                                "2017": 0,
                              },
                            }
          }
/>,
        <Card
          district={
                    Object {
                              "location": "DENVER",
                              "stats": Object {
                                "2017": 0,
                              },
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "COLORADO",
            "stats": Object {
              "2017": 0,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "district": Object {
            "location": "DENVER",
            "stats": Object {
              "2017": 0,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            district={
                        Object {
                                    "location": "COLORADO",
                                    "stats": Object {
                                      "2017": 0,
                                    },
                                  }
            }
/>,
          <Card
            district={
                        Object {
                                    "location": "DENVER",
                                    "stats": Object {
                                      "2017": 0,
                                    },
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "COLORADO",
              "stats": Object {
                "2017": 0,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "district": Object {
              "location": "DENVER",
              "stats": Object {
                "2017": 0,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
